### PR TITLE
Check and fix features

### PR DESCRIPTION
**Pull request by ggwpez**
Update: Trimmed this MR down to only do feature changes and no formatting anymore, to reduce conflicts.

Changes:
- Add Zepter config from the SDK and fix all occurrences
- Add CI check to validate the feature config

[x] Does not require a CHANGELOG entry

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -7,27 +7,27 @@
...
-  "-Aclippy::option-map-unit-fn",        # styalistic
-  "-Aclippy::bind_instead_of_map",       # styalistic
-  "-Aclippy::erasing_op",                # E.g. 0 * DOLLARS
-  "-Aclippy::eq_op",                     # In tests we test equality.
-  "-Aclippy::while_immutable_condition", # false positives
-  "-Aclippy::needless_option_as_deref",  # false positives
-  "-Aclippy::derivable_impls",           # false positives
-  "-Aclippy::stable_sort_primitive",     # prefer stable sort
-  "-Aclippy::extra-unused-type-parameters", # stylistic
+	"-Aclippy::all",
```
Comment (bkontur) [inline .cargo/config.toml:10]: we could also move this to the main Cargo.toml with `[workspace.lints.clippy]` and `[workspace.lints.rust]`,
and set for every Cargo.toml:
```
[lints]
workspace = true
```

**Review by bkontur — APPROVED** [review]

File: .cargo/config.toml (hunk 1 above)
Comment (ggwpez) [reply .cargo/config.toml:10]: Yea agreed, will do in another MR.

**Review by muharem — APPROVED** [review]

**Review by joepetrowski — APPROVED** [review]

**Review by xlc — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

File: .config/zepter.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        '--quiet',
+      ]
+  # Same as `check`, but with the `--fix` flag.
+  default:
+    - [ $check.0, '--fix' ]
+
+# Will be displayed when any workflow fails:
+help:
+  text: |
+    Polkadot-SDK uses the Zepter CLI to detect abnormalities in the feature configuration.
```
Comment (bkchr) [inline .config/zepter.yaml:35]:
```suggestion
    This repo uses the Zepter CLI to detect abnormalities in the feature configuration.
```

File: .config/zepter.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    It looks like one more more checks failed; please check the console output. You can try to automatically address them by running `zepter`.
+    Otherwise please ask directly in the Merge Request, GitHub Discussions or on Matrix Chat, thank you.
+  links:
+    - "https://github.com/paritytech/polkadot-sdk/issues/1831"
```
Comment (bkchr) [inline .config/zepter.yaml:39]:
```suggestion
```

**Review by bkchr — APPROVED** [review]

**Pull request merged**